schema-engine/datamodel-renderer/src/configuration.rs

```diff
@@ -7,38 +7,50 @@ mod generator;
 
 pub use datasource::Datasource;
 pub use generator::Generator;
+use psl::ValidatedSchema;
 
+use std::collections::HashMap;
 use std::fmt;
 
 /// The configuration part of a data model. First the generators, then
 /// the datasources.
 #[derive(Debug, Default)]
 pub struct Configuration<'a> {
-    generators: Vec<Generator<'a>>,
-    datasources: Vec<Datasource<'a>>,
+    /// Generators blocks by file name.
+    pub generators: HashMap<String, Vec<Generator<'a>>>,
```

Comment:
Would it make sense to index them by `FileId` instead, so we won't be passing a bunch of heap allocated strings around?